### PR TITLE
fix(vcl): handle unknown methods

### LINKED ISSUE #790: Non GET requests do not get rejected
https://github.com/adobe/helix-publish/blob/2950616c7afb4eda8fc43dea1018d0021a483123/layouts/fastly/helix.vcl#L1656-L1659

This code, added by @drwilco two years ago does not seem to work (anymore?) Maybe it is because we don't handle the 405 with a custom response?

### PR DESCRIPTION
**Pull request by trieloff**
fixes #790

### CODE REVIEW

File: layouts/fastly/helix.vcl [hunk 1]
```diff
@@ -1119,6 +1119,8 @@ sub hlx_fetch_error {
       error 965 "Too many requests";
     } elsif (beresp.status == 400) {
       error 966 "Bad request";
+    } elsif (beresp.status == 400) {
```
Comment (stefan-guggisberg) [inline layouts/fastly/helix.vcl:1122]: duplicate if stmt

**Review by stefan-guggisberg — CHANGES_REQUESTED** [review]

File: layouts/fastly/helix.vcl (hunk 1 above)
Comment (dominique-pfister) [reply layouts/fastly/helix.vcl:1122]: shouldn't this be 405?

File: layouts/fastly/helix.vcl [hunk 2]
```diff
@@ -1744,7 +1746,12 @@ sub hlx_deliver_errors {
 
   if (resp.status == 966) {
      set resp.status = 400;
-     set resp.response = "Bad request";
+     set resp.response = "Bad Request";
+  }
+
+  if (resp.status == 967) {
+     set resp.status = 400;
```
Comment (dominique-pfister) [inline layouts/fastly/helix.vcl:1753]: and again 405?

**Review by dominique-pfister — APPROVED** [review]

**Pull request merged**